src/main/java/com/es/phoneshop/web/ProductDetailsPageServlet.java

```diff
@@ -13,16 +17,42 @@
 public class ProductDetailsPageServlet extends HttpServlet {
     private ProductDao productDao;
 
+    private CartService cartService;
+
     @Override
     public void init(ServletConfig config) throws ServletException {
         super.init(config);
         productDao = ArrayListProductDao.getInstance();
+        cartService = DefaultCartService.getInstance();
     }
 
     @Override
     protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
         String productId = request.getPathInfo();
         request.setAttribute("product", productDao.getProduct(Long.valueOf(productId.substring(1))));
+        request.setAttribute("cart", cartService.getCart(request));
         request.getRequestDispatcher("/WEB-INF/pages/product.jsp").forward(request, response);
     }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+            Long productId = parseProductId(request);
+            String buttonAdd2Cart = request.getParameter("buttonAdd2Cart");
+            if (request.getParameter("buttonAdd2Cart") != null) {
```

Comment:
I'd recommend to extract strings (at least repeating strings like "buttonAdd2Cart") to constants: Ctrl + Alt + C in IntelliJ IDEA.

This will minimize the chance of making a typo.